tests/Unit/Domain/UserIdTest.php

```diff
@@ -0,0 +1,83 @@
+<?php
+/**
+ * Copyright 2019 SURFnet B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\AzureMfa\Test\Unit\Domain;
+
+
+use PHPUnit\Framework\TestCase;
+use Surfnet\AzureMfa\Domain\UserId;
+use Surfnet\AzureMfa\Domain\Exception\InvalidUserIdException;
+
+class UserIdTest extends TestCase
```

Comment:
The generate, isEqual, getUserId and getEmailAddress methods remain untested. Please add coverage for these methods. The getters could be included in the happy flow test, the isEquals deserves a test method of its own imho.

Reply:
Added a test for the `isEqual` and `generate` method.